### PR TITLE
Start of developmental primitive perceptual representation

### PR DESCRIPTION
**Pull request by rgabbard**
Addresses but does not complete #63

### CODE REVIEW

File: tests/perception/developmental_primitive_perception.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    )
+
+
+def test_relations():
+    # ball on a table
+    ball = ObjectPerception("ball")
+    table = ObjectPerception("table")
+
+    PerceptualRepresentation.single_frame(
+        ObjectPerception(
```
Comment (denizbeser) [inline tests/perception/developmental_primitive_perception.py]: I'm not sure if this test is correct - If I understand the structures correctly, ObjectPerception is for objects (not frames). Should the argument be DevelopmentalPrimitivePerceptionFrame(...) instead?

File: adam/perception/developmental_primitive_perception.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    r"""
+    a set of `ObjectPerception`\ s, with one corresponding to each
+      object in the scene (e.g. a ball, Mom, Dad, etc.)
+    """
+    property_assertions: ImmutableSet["PropertyPerception"] = attrib(
+        converter=_to_immutableset, default=immutableset()
+    )
+    r"""
+    a set of `ObjectPerception`\ s which associate a `ObjectPerception` with perceived properties
+      of various sort (e.g. color, sentience, etc.)
```
Comment (denizbeser) [inline adam/perception/developmental_primitive_perception.py]: Should the documentation here be "a set of 'PropertyPerception'" instead?

File: tests/perception/developmental_primitive_perception.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+
+def test_color():
+    # create a situation with a red ball
+    red = RgbColorPerception(255, 0, 0)
+    ball = ObjectPerception("ball")
+
+    PerceptualRepresentation.single_frame(
+        DevelopmentalPrimitivePerceptionFrame(
+            perceived_objects=[ball], property_assertions=[HasColor(ball, red)]
```
Comment (denizbeser) [inline tests/perception/developmental_primitive_perception.py]: As PyCharm warns I believe the arguments are supposed to be ImmutableSets, so I think we need to convert the lists with ImmutableSet(). The same is true for the arguments in the above test function.

File: tests/perception/developmental_primitive_perception.py (hunk 1 above)
Comment (rgabbard) [reply tests/perception/developmental_primitive_perception.py]: Good catch. This is fixed in 92745a2

File: adam/perception/developmental_primitive_perception.py (hunk 2 above)
Comment (rgabbard) [reply adam/perception/developmental_primitive_perception.py]: Good catch, fixed in c345a2d

File: tests/perception/developmental_primitive_perception.py (hunk 3 above)
Comment (rgabbard) [reply tests/perception/developmental_primitive_perception.py]: This is a PyCharm bug - it does not properly handle the `converter` applies by `attrs`.

**Pull request merged**